### PR TITLE
Use the new internal AWS client

### PR DESCRIPTION
**Pull request by viren-nadkarni**
This PR replaces the use of the old internal AWS client

```
localstack.utils.aws.aws_stack.connect_to_service()
```

with the new one

```
localstack.aws.connect.connect_to()
```

This is part of the plan to remove the legacy internal client entirely since it lacks proper support for multi- and cross- account access.

### CODE REVIEW

File: localstack/aws/forwarder.py [hunk 1]
```diff
@@ -160,12 +160,11 @@ def create_aws_request_context(
...
     # but deactivate validation (validation errors should be handled by the backend)
     # and don't send it yet
-    client = aws_stack.connect_to_service(
-        service_name,
+    client = connect_to(
         endpoint_url=endpoint_url,
         region_name=region,
         config=_non_validating_boto_config,
-    )
+    ).__getattr__(service_name)
```
Comment (dfangl) [inline localstack/aws/forwarder.py:167]: Not a fan of this, for this case we have `connect_to.get_client(`

File: localstack/utils/aws/aws_stack.py [hunk 2]
```diff
@@ -193,10 +194,17 @@ def get_boto3_region() -> str:
     return boto3.session.Session().region_name
 
 
-def is_internal_call_context(headers):
+def is_internal_call_context(headers) -> bool:
```
Comment (dfangl) [inline localstack/utils/aws/aws_stack.py:197]: This is fine for now, preferably we would migrate all this to the is_internal_call property of the RequestContext at some point. Out of scope of this PR of course.

File: localstack/utils/testutil.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ def create_lambda_archive(
 
 
 def delete_lambda_function(name, region_name: str = None):  # TODO: remove all occurrences
-    client = aws_stack.connect_to_service("lambda", region_name=region_name)
+    client = connect_to(region_name=region_name).awslambda
```
Comment (dfangl) [inline localstack/utils/testutil.py:128]: In the test utils we should always use connect_externally_to.

File: tests/integration/test_sqs.py [hunk 4]
```diff
@@ -88,12 +89,11 @@ def test_get_queue_url_contains_request_host(self, sqs_create_queue, monkeypatch
 
         # attempt to connect through a different host and make sure the URL contains that host
         host = f"http://127.0.0.1:{config.EDGE_PORT}"
-        client = aws_stack.connect_to_service(
-            "sqs",
+        client = connect_to(
```
Comment (dfangl) [inline tests/integration/test_sqs.py:92]: Also connect_externally_to please

File: tests/integration/test_dynamodb.py [hunk 5]
```diff
@@ -1570,7 +1571,7 @@ def test_nosql_workbench_localhost_region(self, dynamodb_create_table, aws_clien
         table = aws_client.dynamodb.describe_table(TableName=table_name)
         assert table.get("Table")
         # describe table for "localhost" region
-        client = aws_stack.connect_to_service("dynamodb", region_name="localhost")
+        client = connect_to(region_name="localhost").dynamodb
```
Comment (dfangl) [inline tests/integration/test_dynamodb.py:1574]: Also connect_externally_to please

**Review by dfangl — CHANGES_REQUESTED** [review]
Just some minor comments regarding the clients in test methods and the get attr usage!

File: localstack/aws/forwarder.py (hunk 1 above)
Comment (viren-nadkarni) [reply localstack/aws/forwarder.py]: Changed in aa10884a80270f1bcee30bf94ff8f8341313f3cc :+1:

File: localstack/services/cloudwatch/provider.py [hunk 6]
```diff
@@ -70,7 +71,7 @@ def update_state(target, self, reason, reason_data, state_value):
         data = arns.parse_arn(action)
         # test for sns - can this be done in a more generic way?
         if data["service"] == "sns":
-            service = aws_stack.connect_to_service(data["service"])
+            service = connect_to().__getattr__(data["service"])
```
[the remark this answers is not in the document]
Comment (viren-nadkarni) [reply localstack/services/cloudwatch/provider.py]: Changed in aa10884a80270f1bcee30bf94ff8f8341313f3cc

File: tests/integration/test_dynamodb.py (hunk 5 above)
Comment (viren-nadkarni) [reply tests/integration/test_dynamodb.py]: Good catch, in fact the `aws_client_factory` fixture is the correct thing to use here. Fixed in c18bdc3916650b2f39310b701a6a90c013e57b38

File: tests/integration/test_sqs.py (hunk 4 above)
Comment (viren-nadkarni) [reply tests/integration/test_sqs.py]: Changed in 250caa25fb8336c9293dcd622b4548d151f7ad34.

This is marked for removal anyway, I'll do it in a subsequent PR.

File: localstack/utils/testutil.py (hunk 3 above)
Comment (viren-nadkarni) [reply localstack/utils/testutil.py]: Changed in 250caa25fb8336c9293dcd622b4548d151f7ad34.

This is marked for removal anyway, I'll do it in a subsequent PR.

**Review by dfangl — APPROVED** [review]
LGTM now, thanks for addressing my comments!

**Pull request merged**